include/deal.II/numerics/data_out.h

```diff
@@ -516,6 +516,15 @@ class DataOut : public DataOut_DoFData<DoFHandlerType,
                   const CurvedCellRegion              curved_cell_region);
 };
 
+namespace Legacy
+{
+  /**
+   * @deprecated Use dealii::DataOut without the DoFHandlerType template
+   * instead.
+   */
+  template <int dim, typename DoFHandlerType = DoFHandler<dim>>
+  using DataOut DEAL_II_DEPRECATED = dealii::DataOut<dim, DoFHandlerType>;
```

Comment:
On second though, you mark these names as deprecated. But that means that they will disappear in the following release. But that specifically isn't what we want, do we? Shouldn't these names be currently non-deprecated, be marked as deprecated *after* the upcoming release, and removed after the one after that?

Reply:
As these will get removed at some point, I thought that keeping the deprecation flags would be a good idea. However, it could happen that they'll get accidentally removed for the next release because of this flag. So yes, let's remove them for now.

I'll open a separate draft PR which deals with the deprecation of these aliases, which I will unblock after the next release.

Reply:
Sounds good.